src/projectedTurnout/projectedTurnout.schema.ts

```diff
@@ -6,28 +6,57 @@ import { createZodDto } from 'nestjs-zod'
 
 const ElectionEnum = z.nativeEnum(ElectionCode)
 
-const projectedTurnoutPostSchema = z.object({
-  brPositionId: z.string(),
-  geoid: z.string(),
-  state: z.preprocess(toUpper, z.string()).refine((val) => {
-    if (!val) return true
-    return STATE_CODES.includes(val)
-  }, 'Invalid state code'),
-  L2DistrictType: z.string(),
-  L2DistrictName: z.string(),
+const projectedTurnoutUniqueSchema = z
+  .object({
+    state: z.preprocess(toUpper, z.string()).refine((val) => {
+      if (!val) return true
+      return STATE_CODES.includes(val)
+    }, 'Invalid state code'),
+    L2DistrictType: z.string(),
+    L2DistrictName: z.string(),
+    electionYear: z.preprocess((val) => Number(val), z.number()).optional(),
+    electionDate: z.string().refine((val) => !isNaN(new Date(val).getTime()), {
+      message: 'Invalid date string',
+    }),
+    electionCode: ElectionEnum.optional(),
+  })
+  .refine((data) => data.electionCode || data.electionDate, {
+    message: 'Either electionCode or electionDate is required',
+    path: ['electionCode'],
+  })
 
-  // Numbers
-  electionYear: z.preprocess((val) => Number(val), z.number()),
-  projectedTurnout: z.preprocess((val) => Number(val), z.number()),
+const projectedTurnoutManyQuerySchema = z
+  .object({
+    state: z.preprocess(toUpper, z.string()).refine((val) => {
+      if (!val) return true
+      return STATE_CODES.includes(val)
```

Comment:
Ternary?

Maybe at this point, create an abstracted `validState(val)` schema method to use in all these places?